### ISSUE TITLE
Crashing unikernel does not propagate error return code

### ISSUE THREAD

**Issue by nderjung** [unikraft/kraftkit#701]
### Feature request summary

This issue tracks correctly propagating a return code of `1` when Unikraft/machine crashes. This was discovered by using KraftKit's GitHub Action against `libc-test` which can be viewed here:

https://github.com/nderjung/unikraft/actions/runs/5728684586/job/15523782562#step:5:6723

The above job should have failed because of the crash.

### Related architectures

None

### Related platforms

None

### Additional context

My hunch is that the `Events` method in QEMU's v1alpha1 machine interface implementation is not handling the case where the QEMU binary exits unexpectedly.

**Comment by craciunoiuc** [comment]
After looking over the code for a whole day, I am not sure what happens, but I think it's not something on kraftkit's side.

When the qemu binary is launched, it is started with Parent PID 1, and also the qemu binary returns early (0 exit code), before the the output of the unikernel even starts.

This points me to the idea that when running through QMP, the unikernel is initially started paused and the unix sockets are created. After this is done, the process returns.

Kraftkit then starts the **detached** unikernel qemu process through QMP and communicates with it over there.

This means that, because the processes are not related, we won't be able to fetch the exit code from qemu, and in turn from the Unikernel.

I only looked briefly over the QMP documentation, but I did not find anything related to ExitCodes and such.
Ideally, we could either tell QEMU to write the exit code to a file, or to CLI (not sure if possible), or request the exit code over qmp once it terminates (but we can't because the instance is no longer active). Last option would be to instruct QMP to send over the exit code once it finishes, but again, not sure if such thing exists.

In the meantime, for github-actions I recommend just expecting certain end-lines for success cases and fail on everything else.
